### PR TITLE
Clarify need for theme permission

### LINKED ISSUE #10665: Content bug: webext `theme.getCurrent` and `theme.onUpdated` don't require the theme permission.
## What page(s) did you find the problem on?

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme
https://github.com/mdn/content/blob/573451eba7f31f8e5ef30a354a412bc168318460/files/en-us/mozilla/add-ons/webextensions/api/theme/index.md?plain=1#L15

## Specific page section or heading?

In the main "theme" section, not in any subsection.

## What is the problem?

The documentation says

> To use this API, an extension must request the "theme" permission in its manifest.json file.

This is wrong, the permission is only needed to change the theme with `update()` or `reset()`.
The `getCurrent()` and `onUpdated` APIs are readonly and do not need the permission.

## What did you expect to see?

State which APIs need the permission, and which ones don't. The subpages should mention it too:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme/getCurrent
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme/update
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme/reset
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme/onUpdated

## Did you test this? If so, how?

I tested this by going to about:debugging, debugging an extension which doesn't have the "theme" permission, and trying to use the APIs.

1. Install an add-on without the theme permission if you don't have any. For example, https://addons.mozilla.org/en-US/firefox/addon/tab-counter-plus/
2. Go to `about:debugging#/runtime/this-firefox`
3. Click "inspect" button on the addon
4. In the console, enter `await browser.theme.getCurrent();`
5. You can inspect the object and see that it has the data about your current theme.
6. Then, enter `browser.theme.onUpdated.addListener((theme, windowId) => console.log("Theme updated:", theme))`
7. In a new tab, go to about:addons, choose Themes in the sidebar menu, then enable some theme.
8. Go back to the console, you will see that the new theme has been logged.

Or just go to the source code: https://searchfox.org/mozilla-central/rev/42babdcbeb94de568a1bd78cbdb22593456887ae/toolkit/components/extensions/schemas/theme.json#409,429 . Only the `update` and `reset` APIs have `"permissions": ["theme"]`.

### PR DESCRIPTION
**Pull request by rebloor**
### Description

Clarifies that the theme permission is only required for the update and reset methods. In addition to the confirmation mentioned in the issue, I also confirm this by removing the permission from the manifest of the [theme-integrated-sidebar](https://github.com/mdn/webextensions-examples/tree/main/theme-integrated-sidebar), which uses theme.getCurrent and theme.onUpdated, and confirmed it continued to behave as expected.

Also, tidied up a number of style and grammar issues.

### Related issues and pull requests

Fixes #10665

### CODE REVIEW

File: files/en-us/mozilla/add-ons/webextensions/api/theme/update/index.md [hunk 1]
```diff
@@ -7,7 +7,11 @@ browser-compat: webextensions.api.theme.update
 
 {{AddonSidebar()}}
 
-Updates the browser theme according to the content of given {{WebExtAPIRef("theme.Theme", "Theme")}} object.
+Updates the browser theme according to the content of the {{WebExtAPIRef("theme.Theme", "Theme")}} object.
+
+To use this method, an extension must request the "theme" [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) in its [manifest.json](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json) file.
+
+> **Note:** An extension using the "theme" permission cannot use the manifest [theme](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/theme) key to define a theme.
```
Comment (rpl) [inline files/en-us/mozilla/add-ons/webextensions/api/theme/update/index.md:14]: @rebloor if I recall correctly the actual expeceted behavior is the following:

- only static themes manifest.json files support the "theme" manifest property (and "permissions" or most other properties specific to actual extensions including code are not supported in static theme manifest, it can also be spotted in the API JSONSchema here where we are deriving the theme manifest definition from the base manifest properties + the static themes specifc one: https://searchfox.org/mozilla-central/rev/7d77ff808f8407a3e4fc0911779da446c050f9ee/toolkit/components/extensions/schemas/theme.json#333-362)

- an extension can access the read-only part of the theme API namespace without any additional permission, but to be able to call methods that can change the theme (both update and reset) then the "theme" permission is required (and that is confirmed from the API JSONSchema metadata here: https://searchfox.org/mozilla-central/rev/7d77ff808f8407a3e4fc0911779da446c050f9ee/toolkit/components/extensions/schemas/theme.json#429,449)

it may be worth rewording this note to better match the actual expected behavior (e.g. rewording it the other way around, something like `only static themes manifest.json files can include the "theme" manifest property` with "theme" manifest property linking https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/theme).

Comment (rebloor) [inline files/en-us/mozilla/add-ons/webextensions/api/theme/update/index.md:14]: @rpl I'm a little confused as to why you said

> * an extension can access the read-only part of the theme API namespace without any additional permission, but to be able to call methods that can change the theme (both update and reset) then the "theme" permission is required (and that is confirmed from the API JSONSchema metadata here: https://searchfox.org/mozilla-central/rev/7d77ff808f8407a3e4fc0911779da446c050f9ee/toolkit/components/extensions/schemas/theme.json#429,449)

This was the point of the change and what I now believe is now documented. The link to the schema was also provided in the issue and one of the references I checked.

Comment (rebloor) [inline files/en-us/mozilla/add-ons/webextensions/api/theme/update/index.md:14]: @rpl regarding

> only static themes manifest.json files support the "theme" manifest property (and "permissions" or most other properties specific to actual extensions including code are not supported in static theme manifest, it can also be spotted in the API JSONSchema here where we are deriving the theme manifest definition from the base manifest properties + the static themes specifc one: https://searchfox.org/mozilla-central/rev/7d77ff808f8407a3e4fc0911779da446c050f9ee/toolkit/components/extensions/schemas/theme.json#333-362)

I was trying to unravel the original comment

> When we set up a theme in a background file, we must declare the 'theme' [permission](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions) and therefore we cannot use the [theme](/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/theme) function of the manifest, since it's not compatible.

Unfortunately, I'd forgotten that the theme key is only used to define static themes. I'm thinking it makes sense to remove this note from here (and reset) and rewrite the original note along the lines of:

> You can use this API to include a theme in your extension, which you define as a Theme and apply using the update method. You cannot include a static theme in your extension, defined with the theme manifest key. The theme manifest key is used to define static themes only.

File: files/en-us/mozilla/add-ons/webextensions/api/theme/getcurrent/index.md [hunk 2]
```diff
@@ -22,19 +22,15 @@ let getting = browser.theme.getCurrent(
 ### Parameters
 
 - `windowId` {{optional_inline}}
-  - : `integer`. The ID of a window. If this is provided, the theme applied on that window will be provided. If it is omitted the theme applied on the last focused window will be provided.
+  - : `integer`. The ID of a window. If this is provided, the theme applied on that window is provided. If it is omitted, the theme applied on the last focused window is provided.
```
Comment (rpl) [inline files/en-us/mozilla/add-ons/webextensions/api/theme/getcurrent/index.md:25]: Nit, there is quite some repetition of the work "provided" in this phrase. Not a big deal but would be nice if we could reword it to not repeating it that much, e.g. something like:

> If this is provided, the theme resolved will be the one applied on that window. If it is omitted, the theme resolved will be the one applied to the last focused window.

wdyt?

**Review by rpl — APPROVED** [review]
lgtm, r+wc (where the "with comments" is just a nit described in more detail below)/

File: files/en-us/mozilla/add-ons/webextensions/api/theme/update/index.md (hunk 1 above)
Comment (rpl) [reply files/en-us/mozilla/add-ons/webextensions/api/theme/update/index.md]: This part of my initial comment was just a confirmation related to something that was already correct in the changes included in the PR. No action needed (apologies for the confusion).

Comment (rpl) [reply files/en-us/mozilla/add-ons/webextensions/api/theme/update/index.md]: That reads definitely better. :+1: on reapplying that change.

File: files/en-us/mozilla/add-ons/webextensions/api/theme/getcurrent/index.md (hunk 2 above)
Comment (rebloor) [inline files/en-us/mozilla/add-ons/webextensions/api/theme/getcurrent/index.md:25]:
```suggestion
  - : `integer`. The ID of a window. If this is provided, the theme resolved is the one applied to that window. If it is omitted, the theme resolved is the one applied to the most recently focused window.
```

Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/api/theme/getcurrent/index.md]: Well worth doing - and done - thanks for the prompt review @rpl
